liferay-echo/liferay-portal
# LPS-166635 | LPS-166636 Integration tests for New related assets collection provider - items with same category

## Motivation

Related tickets
[Link to task](https://issues.liferay.com/browse/LPS-166636)
[Link to task](https://issues.liferay.com/browse/LPS-166635)

This PR contains two issues. One, hide the related collection provider under a FF, the other, the integration test for the component


## Change summary:

* Hide items with the same category collection provider behind FF for the epic
* Don't include the selected item in the list of results
* Add integration tests for the component

## How to test the changes, scenario 1

Note: fix for this is in https://github.com/liferay-echo/liferay-portal/pull/10039

* Start the portal with FF LPS-166036 disabled
* Create a new Display Page Template for Web Content
* Add a collection fragment to the page
* Click on the collection fragment to configure it
* Go to the related collection tabs
* Items with the same category collection provider should not be shown


## How to test the changes, scenario 2

* Start the portal with FF LPS-166036 enabled
* Create a new Display Page Template for Web Content
* Add a collection fragment to the page
* Click on the collection fragment to configure it
* Go to the related collection tabs
* Select Assets with the same category collection provider
* Add a heading to the collection fragment and map the title of the web content to it
* Publish the display page template and mark it as default
* Add a new vocabulary
* Add a category to the vocabulary
* Create a new web content and assign the category to it
* Create a second web content and assign the category to it
* Click on the kebab of the first web content and select "View Content"
* You will see the title of the second web in the list

![related](https://user-images.githubusercontent.com/4267448/197755237-7831dae1-788b-4ac8-8b6e-c9744479d13a.gif)
